### DISCUSSION TITLE
Share data/props from API between all components and keep SSG

### DISCUSSION THREAD

**Discussion by kimgiutae** [vercel/next.js#26019] [Help] [answered]
**What is the best way to manage and share external API data between all componets, without loosing static capabilities?**

For example... a lot of my components are using the same data, like product categories. The Header, Footer, Sidebar... even the paths inside categories.

Currently I make a call to the API on each page I need these categories (using getStaticProps), basically everywhere... obviously, this doesn't seems right...

I found in docs that if I use getInitialProps in _app.tsx this disable SSG, and using Context API works only client-side...

**Have you found some solution to this? Or making calls to the API on each page is the right solution right now?**

Thanks.

**Comment by akellbl4** [comment, marked as answer]
Hi @jnthnk,
Fetching all of the data in `getStaticProps` on every page is the right way to do it.
For now Next.js doesn't have any other way to do so.
[Here is dusscussion](https://github.com/vercel/next.js/discussions/10949#discussioncomment-44898) where Tim told they though about shared fetching method but it haven't been introduced yet.

**Reply by kimgiutae** [reply]
Thanks @akellbl4 . This is a very requested featured. Hope the devs introduce this soon...
At least for now we have some solutions, not too elegants, but solutions at least.
